components/list/list-item-drag-handle.js

```diff
@@ -212,52 +167,103 @@ class ListItemDragHandle extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		}));
 	}
 
-	_dispatchActionDown() {
-		this._dispatchAction(dragActions.down);
+	async _doAction(action) {
+		this._dispatchAction(action);
+		const cell = findComposedAncestor(this, (parent) =>  parent.hasAttribute && parent.hasAttribute('draggable'));
+		if (cell) await cell.updateComplete;
+		await this.updateComplete;
+		this.focus();
+		this._movingElement = false;
+	}
+
+	_onDraggerButtonClick() {
+		this.activateKeyboardMode();
 	}
 
-	_dispatchActionUp() {
-		this._dispatchAction(dragActions.up);
+	_onDraggerButtonKeydown(e) {
+		if (e.keyCode !== keyCodes.ENTER && e.keyCode !== keyCodes.SPACE) return;
+		e.preventDefault();
+		this.activateKeyboardMode();
 	}
 
-	async _onActiveKeyboard(e) {
-		if (!this._keyboardActive) {
+	_onKeyboardButtonFocusIn() {
+		if (hasDisplayedKeyboardTooltip) return;
+		this._displayKeyboardTooltip = true;
+		hasDisplayedKeyboardTooltip = true;
+	}
+
+	_onKeyboardButtonFocusOut(e) {
+		this._displayKeyboardTooltip = false;
+		if (this._movingElement) {
+			this._movingElement = false;
+			e.stopPropagation();
+			e.preventDefault();
 			return;
 		}
+		this._keyboardActive = false;
+		this._dispatchAction(dragActions.save);
+		e.stopPropagation();
+	}
+
+	async _onMoveButtonAction(e) {
+
 		let action = null;
-		switch (e.keyCode) {
-			case keyCodes.UP:
+		switch (e.detail.action) {
+			case moveActions.up:
 				this._movingElement = true;
 				action = dragActions.up;
 				this.updateComplete.then(() => this.blur()); // tell screenreaders to refocus
 				break;
-			case keyCodes.DOWN:
+			case moveActions.down:
 				this._movingElement = true;
 				action = dragActions.down;
 				break;
-			case keyCodes.HOME:
+			case moveActions.right:
+				this._movingElement = true;
+				action = (this.dir === 'rtl' ? dragActions.unnest : dragActions.nest);
+				break;
+			case moveActions.left:
+				this._movingElement = true;
+				action = (this.dir === 'rtl' ? dragActions.nest : dragActions.unnest) ;
+				break;
+			case moveActions.rootHome:
+				this._movingElement = true;
+				action = dragActions.rootFirst;
+				break;
+			case moveActions.home:
+				this._movingElement = true;
+				action = dragActions.first;
+				break;
+			case moveActions.rootEnd:
 				this._movingElement = true;
-				action = (e.ctrlKey ? dragActions.rootFirst : dragActions.first);
+				action = dragActions.rootLast;
 				break;
-			case keyCodes.END:
+			case moveActions.end:
 				this._movingElement = true;
-				action = (e.ctrlKey ? dragActions.rootLast : dragActions.last);
+				action = dragActions.last;
 				break;
+			default:
+				return;
+		}
+
+		this._doAction(action);
+
+	}
+
+	async _onMoveButtonKeydown(e) {
```

Comment:
While the above `_onMoveButtonAction` handler maps the move event `moveActions` to `dragActions` that list and list consumers understand, the drag-handle still needs to handle key events for tab, escape, enter, and space as it previously did.